test/engine-tap/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+make_lua_path(LUA_PATH
+  PATHS
+    ${CMAKE_CURRENT_SOURCE_DIR}/lib/?.lua
+)
+
+set(TEST_SUITE_NAME "engine-tap")
+set(TARANTOOL_BIN $<TARGET_FILE:tarantool>)
+
+message(STATUS "Add test suite ${TEST_SUITE_NAME}")
+
+# XXX: The call produces both test and target <engine-tap-deps>
+# as a side effect.
+_add_test_suite_target(${TEST_SUITE_NAME}
+  LABELS ${TEST_SUITE_NAME}
+)
+
+list(APPEND TEST_ENV "LUA_PATH=${LUA_PATH}")
+
+file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *.test.lua)
+foreach(test_path ${tests})
+  get_filename_component(test_name ${test_path} NAME)
+  # FIXME: By default, GLOB lists directories.
+  # Directories are omitted in the result if LIST_DIRECTORIES
+  # is set to false. New in version CMake 3.3.
+  if(${test_name} STREQUAL ${TEST_SUITE_NAME})
+    continue()
+  endif()
+  set(test_title "test/${TEST_SUITE_NAME}/${test_name}")
+  set(work_dir "${CMAKE_CURRENT_BINARY_DIR}/${test_name}-workdir")
+  file(MAKE_DIRECTORY ${work_dir})
+  set(TEST_CMD "${RM} -f ${work_dir}/* && ${TARANTOOL_BIN} ${test_path}")
+  add_test(NAME ${test_title}
+           COMMAND ${BASH} -c "${TEST_CMD}"
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+  list(APPEND TEST_ENV "TT_WORK_DIR=${work_dir}")
+  set_tests_properties(${test_title} PROPERTIES
+    ENVIRONMENT "${TEST_ENV}"
+    LABELS "${TEST_SUITE_NAME};integration"
+    DEPENDS ${TEST_SUITE_NAME}-deps
+    FAIL_REGULAR_EXPRESSION "${CTEST_FAIL_REGEXP}"
+  )
+endforeach()
+
+# XXX: Parameterize tests, test/engine-tap/engine.cfg.
+# "memtx": {"engine": "memtx"},
+# "vinyl": {"engine": "vinyl"}
+
+# XXX: MEMTX_ALLOCATOR in ./engine-tap/lib/tester.lua.
```

Comment:
Friendly reminder to update parametrization.

Reply:
I'll add ctest support for TAP tests in a separate PR.